### PR TITLE
Added dirty_cat as a package.

### PR DESCRIPTION
**Pull request by thewchan**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by thewchan** [comment]
@conda-forge/staged-recipes ready for review!

File: recipes/dirty_cat/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - numpy >=1.16
+    - pandas >=1.1.5
+    - python >=3.6
+    - requests
+    - scikit-learn >=0.21
+    - scipy >=1.2
+
+test:
+  imports:
+    - benchmarks
```
Comment (dopplershift) [inline recipes/dirty_cat/meta.yaml:33]: This means the package is installing its benchmarks as a globally importable `benchmarks` package--which could quite easily conflict, and doesn't seem useful. Can you remove the source before install or hack the `setup.py`?

Comment (thewchan) [reply recipes/dirty_cat/meta.yaml:33]: I'll give it a try with a patch

File: recipes/dirty_cat/0001-remove-benchmark.patch [hunk 2]
```diff
@@ -0,0 +1,11 @@
+--- setup.py	2021-12-19 11:54:50.000000000 -0500
++++ setup.py	2022-01-10 17:34:31.594365500 -0500
+@@ -36,7 +36,7 @@
+               'Topic :: Software Development :: Libraries',
+           ],
+           platforms='any',
+-          packages=find_packages(),
++          packages=find_packages(exclude=["benchmark"]),
```
Comment (dopplershift) [inline recipes/dirty_cat/0001-remove-benchmark.patch:8]: "benchmarks"

**Comment by dopplershift** [comment]
Since this is still passing, that means the patch isn't working, likely due to spelling as I commented above.

File: recipes/dirty_cat/0001-remove-benchmark.patch (hunk 2 above)
Comment (thewchan) [reply recipes/dirty_cat/0001-remove-benchmark.patch:8]: whoops. Yeah also I guess I need to take out the `import benchmarks test`

**Comment by thewchan** [comment]
@dopplershift I think this is good to go?

File: recipes/dirty_cat/0001-remove-benchmarks.patch [hunk 3]
```diff
@@ -0,0 +1,11 @@
+--- setup.py	2021-12-19 11:54:50.000000000 -0500
++++ setup.py	2022-01-10 17:34:31.594365500 -0500
+@@ -36,7 +36,7 @@
+               'Topic :: Software Development :: Libraries',
+           ],
+           platforms='any',
+-          packages=find_packages(),
++          packages=find_packages(exclude=["benchmarks"]),
```
Comment (dopplershift) [inline recipes/dirty_cat/0001-remove-benchmarks.patch:8]: Can you change this to explicitly only list `dirty_cat` instead? The source tarball from PyPI has an additional `src` directory that is installed globally and isn't used by the package itself anywhere.

Comment (thewchan) [reply recipes/dirty_cat/0001-remove-benchmarks.patch]: yep let me try

**Comment by thewchan** [comment]
@dopplershift how's it looking now?

**Review by dopplershift — APPROVED** [review]

**Pull request merged**